src/machine/state/mod.rs

```diff
@@ -16,6 +17,49 @@ pub struct State {
 
 impl State {
     pub fn new() -> Self {
-        unimplemented!();
+        State {
+            // for the moment start execution at address 0
+            pc: 0,
+
+            sr: SRegisters::new(),
+            gpr: GPRegisters::new(),
+
+            // for the moment create 1MB of memory
+            mem: Memory::with_capacity(0x100000)
+        }
+    }
+
+    /// Display the state partially on the command line for testing purposes
+    pub fn debug_output(&self) {
```

Comment:
To your first problem: Why would I implement something now when I know exactly that I will need to change everything again tomorrow? 

To your suggestion: Sounds really fancy, but do you think this can be implemented until Thursday?

Suggestion: We first merge this PR, so that we can actually execute and see stuff happening and debug. Then, if we have still got time we could implement your suggestion.